### PR TITLE
Snow issue402

### PR DESCRIPTION
**Pull request by lrochette**
This is a big set of changes:
 - #402: introduce closeCR action to close an existing CR
 - added updateCR action to modify an existing CR  (same API than close but with different body basically)
 - URL to the CR created now shows up on the creation step
 - more error checks about expected values
 - sample in the step.yaml updated
 - Breaking changes: parameters have been renamed to:
   - be all uppercase
   - have CR_ (or SN_) prefix for  consistency

### CODE REVIEW

File: incubating/service-now/step.yaml [hunk 1]
```diff
@@ -1,8 +1,8 @@
-kind: step-type
+quay.io/codefreshplugins/kind: step-type
```
Comment (TedSpinks) [inline incubating/service-now/step.yaml:1]: looks like maybe a find/replace artifact?

File: incubating/service-now/step.yaml [hunk 2]
```diff
@@ -17,7 +17,7 @@ metadata:
   tags: []
   icon:
     type: svg
-    url: https://raw.githubusercontent.com/codefresh-io/steps/master/incubating/service-now/images/servicenow-1.png
```
Comment (TedSpinks) [inline incubating/service-now/step.yaml:20]: I think this will need to be 'master' once its published, right?

Comment (lrochette) [reply incubating/service-now/step.yaml:20]: yes

File: incubating/service-now/step.yaml [hunk 3]
```diff
@@ -27,11 +27,44 @@ metadata:
...
+            duration: 48
+            timeUnit: hours
+            finalState: denied
+          modifyCR:
+            title: Close the CR
+            type: service-now
+            stage: deploy
+            arguments:
+              CR_ACTION: closeCR
+              CR_SYSID: ${{CR_SYSID}}'
```
Comment (dustinvanbuskirk) [inline incubating/service-now/step.yaml:49]: Missing single quote from beginning.

File: incubating/service-now/step.yaml [hunk 4]
```diff
@@ -27,11 +27,44 @@ metadata:
...
+              SN_INSTANCE: https://instance.service-now.com
+              SN_USER: ${{SN_USER}}
+              SN_PASSWORD: ${{SN_PASSWORD}}
+              CR_DATA: {"test_plan":"We ran all the test in the awesome test suite. Test pass rate: ${{TEST_RATE}}"}
```
Comment (dustinvanbuskirk) [inline incubating/service-now/step.yaml:53]: CR_DATA should be single quoted in example or bad parsing will take place.

File: incubating/service-now/step.yaml [hunk 5]
```diff
@@ -27,11 +27,44 @@ metadata:
...
+          stage: cleaning
+          arguments:
+            CR_ACTION: closeCR
+            CR_SYSID: ${{CR_SYSID}}
+            SN_INSTANCE: https://instance.service-now.com
+            SN_USER: ${{SN_USER}}
+            SN_PASSWORD: ${{SN_PASSWORD}}
+            CR_CLOSE_CODE: successful
+            CR_CLOSE_NOTES: "CR closed by Codefresh build ${{CF_BUILD_ID}}"
+            CR_DATA: {"work_notes":"Closed automatically by Codefresh build ${{CF_BUILD_ID}}"}
```
Comment (dustinvanbuskirk) [inline incubating/service-now/step.yaml:66]: CR_DATA should be single quoted in example or bad parsing will take place.

File: incubating/service-now/step.yaml (hunk 4 above)
Comment (lrochette) [reply incubating/service-now/step.yaml:53]: Fixed in the next version

**Pull request merged**